src/components/Form/Form.tsx

```diff
@@ -0,0 +1,133 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import { UIComponent, childrenExist, customPropTypes } from '../../lib'
+import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
+import {
+  ComponentEventHandler,
+  Extendable,
+  ReactChildren,
+  ShorthandValue,
+} from '../../../types/utils'
+import { Sizes } from '../../lib/enums'
+import Grid from '../Grid/Grid'
+import FormField from './FormField'
+
+export interface IFormProps {
+  action?: string
+  as?: any
+  children?: ReactChildren
+  className?: string
+  onSubmit?: ComponentEventHandler<IFormProps>
+  columns?: string | number
+  content?: ShorthandValue | ShorthandValue[]
+  rows?: string | number
+  size?: Sizes
+  loading?: boolean
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A Form displays a set of related user input fields in a structured way.
+ */
+class Form extends UIComponent<Extendable<IFormProps>, any> {
+  static create: Function
+
+  public static displayName = 'Form'
+
+  public static className = 'ui-form'
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /**
+     *  Button content for childrenApi
+     *  @docSiteIgnore
+     */
+    children: PropTypes.node,
+
+    /** Additional CSS class name(s) to apply.  */
+    className: PropTypes.string,
+
+    /** The columns of the grid with a space-separated list of values. The values represent the track size, and the space between them represents the grid line. */
+    columns: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+    /** Shorthand for primary content. */
+    content: customPropTypes.every([
+      customPropTypes.disallow(['children']),
+      PropTypes.oneOfType([
+        PropTypes.arrayOf(customPropTypes.itemShorthand),
+        customPropTypes.itemShorthand,
+      ]),
+    ]),
+
+    /** The rows of the grid with a space-separated list of values. The values represent the track size, and the space between them represents the grid line. */
+    rows: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+    /**
+     * The HTML form submit handler.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onSubmit: PropTypes.func,
+
+    /** Automatically show a loading indicator. */
+    loading: PropTypes.bool,
```

Comment:
emm, not sure that it should be part of the default form interface. To me `loading` seems to be something that introduces behavioural aspects to the form - and, thus, see this `loading` aspect to be used as rather _decorator_ for the general `Form` component.

Reply:
Make sense, will remove this prop for now. It seems like a good concept, but in the end introducing this here would then bring out question as why only this component has the loading indicator, so for this moment I agree that it should not be part of the Form component.